server/src/core/python_arch_eval.rs

```diff
@@ -1096,11 +1117,11 @@ impl PythonArchEval {
             for decorator_eval in dec_evals.iter(){
                 let EvaluationSymbolPtr::WEAK(decorator_eval_sym_weak) = decorator_eval.symbol.get_symbol(session, &mut None, &mut self.diagnostics, None)  else {continue};
                 let Some(dec_sym) = decorator_eval_sym_weak.weak.upgrade() else {continue};
-                let dec_sym_tree = dec_sym.borrow().get_tree();
-                if dec_sym_tree == (vec![yarn!("odoo"), yarn!("api")], vec![yarn!("returns")]){
-                    PythonArchEval::handle_api_returns_decorator(session, func_sym.clone(), decorator_args);
-                } else if dec_sym_tree == (vec![yarn!("odoo"), yarn!("api")], vec![yarn!("onchange")]) ||
-                        dec_sym_tree == (vec![yarn!("odoo"), yarn!("api")], vec![yarn!("constrains")]){
+                let dec_sym_tree = dec_sym.borrow().get_main_entry_tree(session);
```

Comment:
Try to not call get_main_entry_tree if possible, maybe do basics checks first

Reply:
Updated